src/compiler/checker.ts

```diff
@@ -21301,18 +21303,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     if (result === Ternary.True || result === Ternary.Maybe) {
                         // If result is definitely true, record all maybe keys as having succeeded. Also, record Ternary.Maybe
                         // results as having succeeded once we reach depth 0, but never record Ternary.Unknown results.
-                        for (let i = maybeStart; i < maybeCount; i++) {
-                            relation.set(maybeKeys[i], RelationComparisonResult.Succeeded | propagatingVarianceFlags);
-                        }
+                        maybeKeys.popAll(v => {

```

Comment:
Any idea how many of these typically get popped at a time?

Reply:
No but I can see if I can collect that info.

Reply:
The latest version no longer requires a call each time, though in this case it's required. What optimization were you thinking of?

Reply:
If you typically have very long strings of pushes and pops, you could even get rid of the explicit array and just iterate over the keys of the map.

*Ideally*, we could iterate from the end and bail out and get rid of the array entirely. See https://github.com/tc39/notes/blob/67ba44cc70f3338a2b8dcecd5eebec26ba5e9a40/meetings/2019-07/july-23.md#symbolreverse